### PR TITLE
add peer dependency for webpack 4 types

### LINKED ISSUE #3463: [heft] using both heft plugins for webpack4 and webpack5 in a project can cause webpack type errors
## Summary
typescript will throw errors when you have both webpack 5 and webpack 4 rigs in a project 

## Repro steps
##[error][typescript] Error: ../../../common/temp/node_modules/.pnpm/@rushstack+module-minifier-plugin@0.9.2_8a934206096b177ef4e8c472548f577c/node_modules/@rushstack/module-minifier-plugin/dist/module-minifier-plugin.d.ts:17:10 - (TS2305) Module '"webpack"' has no exported member 'Plugin'.
##[error][typescript] Error: ../../../common/temp/node_modules/.pnpm/@rushstack+module-minifier-plugin@0.9.2_8a934206096b177ef4e8c472548f577c/node_modules/@rushstack/module-minifier-plugin/dist/module-minifier-plugin.d.ts:39:74 - (TS2724) 'exports' has no exported member named 'compilation'. Did you mean 'Compilation'?

  **Expected result:** 

  **Actual result:** 

## Details
Adding @types/webpack as a peer dependency in heft-webpack4-plugin should solve this problem

### PR DESCRIPTION
**Pull request by TheLarkInn**
Fixes #3463

### CODE REVIEW

File: heft-plugins/heft-webpack4-plugin/package.json [hunk 1]
```diff
@@ -19,6 +19,7 @@
   },
   "peerDependencies": {
     "@rushstack/heft": "^0.45.6",
+    "@types/webpack": "4.41.24",
```
Comment (dmichon-msft) [inline heft-plugins/heft-webpack4-plugin/package.json:22]: Also:
```
"peerDependenciesMeta": {
  "@types/webpack": {
    "optional": true
  }
},
```

**Review by dmichon-msft — CHANGES_REQUESTED** [review]
Needs to be an optional peer dependency for non-typescript consumers

File: heft-plugins/heft-webpack4-plugin/package.json (hunk 1 above)
Comment (dmichon-msft) [reply heft-plugins/heft-webpack4-plugin/package.json:22]: Also, this should be some form of range; we don't want to impose exact versions in a peer dependency

Comment (TheLarkInn) [reply heft-plugins/heft-webpack4-plugin/package.json]: Fixed. Using ^4 as the type range

**Comment by iclanton** [comment]
You need to run `rush change`.

**Review by iclanton — APPROVED** [review]

**Review by dmichon-msft — APPROVED** [review]

**Comment by TheLarkInn** [comment]
Done @iclanton

**Pull request merged**